third_party/xla/xla/stream_executor/gpu/gpu_cudamallocasync_allocator.cc

```diff
@@ -215,6 +215,15 @@ GpuCudaMallocAsyncAllocator::GpuCudaMallocAsyncAllocator(
   static auto* all_pools_ = new std::vector<CUmemoryPool*>();
   static auto* all_ids_ = new std::vector<tsl::PlatformDeviceId>();
   DCHECK(all_pools_->size() == all_ids_->size());
+  for (auto& pool_item_ : *all_pools_) {
+    if (*pool_item_ == pool_) {
```

Comment:
Just to check my understanding - each device has a default memory pool. Virtual devices on the same physical device will use the same memory pool. So for all virtual devices on the same physical device, we only initialize the pool of this physical device once. In other words, if the `PlatformDeviceId` already appears in the `all_ids_`, the pool initialization can be skipped. Right?

Can you add some comment for this?

Reply:
Yes, your understanding is correct. Add comments ahead.